benchmarks/queue/src/main.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use clap::Parser;
+use iceoryx2_bb_lock_free::spsc::index_queue::FixedSizeIndexQueue;
+use iceoryx2_bb_lock_free::spsc::queue::Queue;
+use iceoryx2_bb_lock_free::spsc::safely_overflowing_index_queue::FixedSizeSafelyOverflowingIndexQueue;
+use iceoryx2_bb_posix::barrier::*;
+use iceoryx2_bb_posix::clock::Time;
+use iceoryx2_bb_posix::thread::ThreadBuilder;
+
+const ITERATIONS: u64 = 100000000;
+
+trait PushPop: Send + Sync {
+    fn push(&self, value: usize);
+    fn pop(&self) -> Option<usize>;
+}
+
+impl<const CAPACITY: usize> PushPop for Queue<usize, CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(&value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+impl<const CAPACITY: usize> PushPop for FixedSizeIndexQueue<CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+impl<const CAPACITY: usize> PushPop for FixedSizeSafelyOverflowingIndexQueue<CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+fn perform_benchmark<Q: PushPop>(
+    args: &Args,
+    queue_a2b: Q,
+    queue_b2a: Q,
```

Comment:
These do not need to be passed but could be created right in the function

Reply:
But I wanted to pass them so that we can add later also relocatable queues that are more complex to construct. Or queues where the capacity is not a compile time arg but a runtime new arg